### PR TITLE
compiler error fix

### PR DESCRIPTION
**Pull request by AlphaLima**
changed logical to bitwise AND ( compiler error fix)
always use APT buffer as target from LK destination.
allow STF switch bi-directional exchange

### CODE REVIEW

File: Common/Source/Devices/devLX_EOS_ERA.cpp [hunk 1]
```diff
@@ -1983,15 +1984,11 @@ TCHAR szName[MAX_VAL_STR_LEN];
   {                                  
     int rwdir = 0;
```
Comment (brunotl) [inline Common/Source/Devices/devLX_EOS_ERA.cpp:1985]: always "0" ...

File: Common/Source/Devices/devLX_EOS_ERA.cpp [hunk 2]
```diff
@@ -405,6 +405,7 @@ int PortNum = d->PortNumber;
     dfe->addEnumText(MsgToken(491)); // LKTOKEN  _@M491_ "OFF"
     dfe->addEnumText(MsgToken(2452)); // LKTOKEN  _@M2452_ "IN"
     dfe->addEnumText(MsgToken(2453)); // LKTOKEN  _@M2453_ "OUT"
+    dfe->addEnumText(MsgToken(2454)); // LKTOKEN  _@M2454_ "IN & OUT"
```
Comment (brunotl) [inline Common/Source/Devices/devLX_EOS_ERA.cpp:408]: unrelated to change in "DevLX_EOS_ERA::PutTarget"

Comment (AlphaLima) [reply Common/Source/Devices/devLX_EOS_ERA.cpp:408]: right! should be related to STF switch!

Comment (AlphaLima) [reply Common/Source/Devices/devLX_EOS_ERA.cpp:408]: that is correct as it is it is intendend to have the possibility to trigger STF from the device and LK.

File: Common/Source/Devices/devLX_EOS_ERA.cpp (hunk 1 above)
Comment (AlphaLima) [reply Common/Source/Devices/devLX_EOS_ERA.cpp:1985]: ... that is correct in case it is no landaable
changed it in next commit for lanables
